_blogposts/2023-05-17-enhancing-the-ergonomics-of-records.mdx

```diff
@@ -0,0 +1,140 @@
+---
+author: rescript-team
+date: "2023-05-17"
+title: Enhancing the ergonomics of records
+badge: roadmap
+description: |
+  A tour of new capabilities coming in ReScript v11
+---
+
+[Records](https://rescript-lang.org/docs/manual/latest/record) are a fundamental part of ReScript, representing an efficient and expressive way to model structured data. Beyond offering a clear and concise definition of complex data structures, records bring numerous advantages such as strong type checking, immutability by default, great error messages, and support for exhaustive pattern matching. This allows you to write robust and maintainable code, while the compiler ensures you've handled all possible cases when dissecting a record.
```

Comment:
A few points here:
I think most folks should be familiar with records, so no need to advertise them so heavily.

I assume the major message here is that there are records, and structurally typed objects. The latter is more convenient to use due to their ability for subtyping. Records didn't have that ergonomics so fare, but they are generally nicer to use due to its nominal typing.

To encourage more record usage, we made the way of defining records and coercing (sub-typing) records easier so we don't have to retreat to structured objects?

Reply:
Check again now, I rewrote parts of the intro.